lib/MetaCPAN/Model/Search.pm

```diff
@@ -0,0 +1,484 @@
+package MetaCPAN::Model::Search;
+
+use Moose;
```

Comment:
Any reason not to use `MetaCPAN::Moose` here?  That gives you `MooseX::StrictConstructor` for free.